src/Compilers/Core/Portable/InternalUtilities/StringTable.cs

```diff
@@ -730,25 +736,7 @@ internal static unsafe bool TextEqualsASCII(string text, byte* ascii, int length
             return true;
         }
 
-        internal static bool TextEquals(string array, char[] text, int start, int length)
-        {
-            return array.Length == length && TextEqualsCore(array, text, start);
-        }
-
-        private static bool TextEqualsCore(string array, char[] text, int start)
-        {
-            // use array.Length to eliminate the range check
-            int s = start;
-            for (var i = 0; i < array.Length; i++)
-            {
-                if (array[i] != text[s])
-                {
-                    return false;
-                }
-                s++;
-            }
-
-            return true;
-        }
+        internal static bool TextEquals(string array, ReadOnlySpan<char> text)
+            => text.Equals(array.AsSpan(), StringComparison.Ordinal);
```

Comment:
>text.Equals [](start = 15, length = 11)

I didn't understand how this `.Equals` works. 
`ReadOnlySpan.Equals` seems to throw an exception ([source](https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/ReadOnlySpan.cs#L63))

Reply:
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs#L37

Reply:
Thanks!